src/main/java/org/cbio/gdcpipeline/writer/ClinicalPatientDataWriter.java

```diff
@@ -0,0 +1,179 @@
+
+    private String dataOutputFile;
+
+    private Resource resource;
+
+
+    @Value("${clinical.data.patient.file.name}")
+    private String CLINICAL_DATA_PATIENT_FILE_NAME;
+
+    @Value("${clinical.metadata.patient.file.name}")
+    private String CLINICAL_METADATA_PATIENT_FILE_NAME;
+
+    FlatFileItemWriter<PatientFileModel> dataWriter = new FlatFileItemWriter<>();
+
+    @Autowired
+    ClinicalPatientFileHeaders patientFileHeaders;
+
+
+    private static Log LOG = LogFactory.getLog(ClinicalPatientDataWriter.class);
+
+    private String modifyRow(String[] row) {
+        // Tab Delimited
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < row.length; i++) {
+            if (i == 0) {
+                sb.append(row[i]);
+            } else {
+                sb.append("\t");
+                sb.append(row[i]);
+            }
+        }
+        return sb.toString();
+    }
+
+
+    private FlatFileHeaderCallback clinicalDataHeader() {
+        return new FlatFileHeaderCallback() {
+            @Override
+            public void writeHeader(Writer writer) throws IOException {
+
+
+                StringBuilder sb = new StringBuilder();
+                sb.append(modifyRow(patientFileHeaders.getRow1()));
+                sb.append(System.lineSeparator());
+                sb.append(modifyRow(patientFileHeaders.getRow2()));
+                sb.append(System.lineSeparator());
+                sb.append(modifyRow(patientFileHeaders.getRow3()));
+                sb.append(System.lineSeparator());
+                sb.append(modifyRow(patientFileHeaders.getRow4()));
+                sb.append(System.lineSeparator());
+                sb.append(modifyRow(patientFileHeaders.getRow5()));
+                //sb.append(System.lineSeparator());
+
+                writer.write(sb.toString());
+
+            }
+        };
+    }
+
+    public FieldExtractor<PatientFileModel> createFieldExtractor() {
+        BeanWrapperFieldExtractor<PatientFileModel> ext = new BeanWrapperFieldExtractor<>();
+
+        String[] fields = patientFileHeaders.getRow5().clone();
+
+        for (int i = 0; i < fields.length; i++) {
+            fields[i] = fields[i].toLowerCase();
+        }
+        ext.setNames(fields);
+        return ext;
+    }
+
+    @Override
+    public void open(ExecutionContext executionContext) throws ItemStreamException {
+
+        //configure writer
+        dataWriter.setAppendAllowed(true);
+        dataWriter.setShouldDeleteIfExists(true);
+        dataWriter.setHeaderCallback(clinicalDataHeader());
+        dataWriter.setLineSeparator(System.lineSeparator());
+
+        DelimitedLineAggregator<PatientFileModel> lineAggregator = new DelimitedLineAggregator<>();
+        lineAggregator.setDelimiter("\t");
+        FieldExtractor<PatientFileModel> fe = createFieldExtractor();
+        lineAggregator.setFieldExtractor(fe);
+        dataWriter.setLineAggregator(lineAggregator);
+
+        if (outputDir.isEmpty()) {
```

Comment:
If no destination is specified, the job should have exited in the main method in the GDCPipelineApplication class.